cognite/neat/graph/transformations/query_generator/sparql.py

```diff
@@ -283,3 +290,246 @@ def build_sparql_query(
+    -------
+    tuple[list[Triple],list[Triple]]
+        _description_
+    """
+
+    templates = []
+    patterns = []
+
+    # here pull all properties which are of type `rdfpath` and are defined for the class:
+    properties = [
+        property_.model_copy()
+        for property_ in get_classes_with_properties(transformation_rules)[class_]
+        if property_.rule_type == RuleType.rdfpath and not property_.skip_rule
+    ]
+
+    # TODO: Add handling of UNIONs in rules
+
+    # parse rules for those properties
+    for i, property_ in enumerate(properties):
+        properties[i].rule = parse_rule(property_.rule, property_.rule_type).traversal
+
+    # add first triple for graph pattern stating type of object
+    patterns += [
+        Triple(
+            subject="?subject",
+            predicate="a",
+            object=_most_occurring_element([property_.rule.class_.id for property_ in properties]),
+            optional=False,
+        )
+    ]
+
+    for property_ in properties:
+        graph_template_triple = Triple(
+            subject="?subject",
+            predicate=f"{transformation_rules.metadata.prefix}:{property_.property_id}",
+            object=f'?{re.sub(r"[^_a-zA-Z0-9/_]", "_", str(property_.property_id).lower())}',
+            optional=False,
+        )
+
+        if isinstance(property_.rule, AllReferences):
+            graph_pattern_triple = Triple(
+                subject="BIND(?subject",
+                predicate="AS",
+                object=f"{graph_template_triple.object})",
+                optional=True if properties_optional else not property_.mandatory,
+            )
+
+        elif isinstance(property_.rule, SingleProperty):
+            graph_pattern_triple = Triple(
+                subject=graph_template_triple.subject,
+                predicate=property_.rule.property.id,
+                object=graph_template_triple.object,
+                optional=True if properties_optional else not property_.mandatory,
+            )
+
+        elif isinstance(property_.rule, Hop):
+            graph_pattern_triple = Triple(
+                subject="?subject",
+                predicate=_hop2property_path(graph, property_.rule, transformation_rules.prefixes),
+                object=graph_template_triple.object,
+                optional=True if properties_optional else not property_.mandatory,
+            )
+        else:
+            continue
+
+        patterns += [graph_pattern_triple]
+        templates += [graph_template_triple]
+
+    return templates, patterns
+
+
+def _count_element_occurrence(list_of_elements: list):
+    occurrence: dict[str, int] = {}
+    for i in list_of_elements:
+        if i in occurrence:
+            occurrence[i] += 1
+        else:
+            occurrence[i] = 1
+    return occurrence
+
+
+def _most_occurring_element(list_of_elements: list):
+    occurrence = _count_element_occurrence(list_of_elements)
+    return max(occurrence, key=occurrence.get)
```

Comment:
This comes built in to Python, I  think it should be:
```suggestion
def _most_occurring_element(list_of_elements: list):
    counts = Counter(list_of_elements)
    return counts.most_common(1)[0][0]
```

Reply:
sweat !